### PR TITLE
Introduce convert methods

### PR DESCRIPTION
**Pull request by shogo82148**
## Summary by CodeRabbit

- **New Features**
  - Added support for seamless conversion between various custom fixed-width signed and unsigned integer types, from Int8 through Int1024 and Uint8 through Uint1024.

- **Tests**
  - Introduced comprehensive tests validating integer type conversions across all supported types, ensuring accuracy for diverse values and edge cases.

### CODE REVIEW

File: cast_test.go [hunk 1]
```diff
@@ -0,0 +1,1975 @@
...
+			},
+			-1,
+		},
+	}
+
+	for _, tc := range testCases {
+		got := tc.a.Int32()
+		if got != tc.want {
+			t.Errorf("Int1024(%#0256x).Int16() = %#08x, want %#08x", tc.a, got, tc.want)
+		}
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply cast_test.go:1466]: fixed in https://github.com/shogo82148/ints/pull/29/commits/d0cff8efba17da8a97f12d83c9e0ce5534073a52

File: cast_test.go [hunk 2]
```diff
@@ -0,0 +1,1975 @@
...
+			},
+			-1,
+		},
+	}
+
+	for _, tc := range testCases {
+		got := tc.a.Int64()
+		if got != tc.want {
+			t.Errorf("Int1024(%#0256x).Int16() = %#016x, want %#016x", tc.a, got, tc.want)
+		}
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply cast_test.go:1503]: fixed in https://github.com/shogo82148/ints/pull/29/commits/d0cff8efba17da8a97f12d83c9e0ce5534073a52

File: cast_test.go [hunk 3]
```diff
@@ -0,0 +1,1975 @@
...
+				0, 0, 0, 0, 0, 0, 0, 1,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		got := tc.a.Int1024()
+		if got != tc.want {
+			t.Errorf("Int1024(%#0256x).Int16() = %#0256x, want %#0256x", tc.a, got, tc.want)
+		}
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply cast_test.go:1656]: fixed in https://github.com/shogo82148/ints/pull/29/commits/d0cff8efba17da8a97f12d83c9e0ce5534073a52

File: cast_test.go [hunk 4]
```diff
@@ -0,0 +1,1975 @@
...
+			},
+			Int128{M - 1, M - 1},
+		},
+	}
+
+	for _, tc := range testCases {
+		got := tc.a.Int128()
+		if got != tc.want {
+			t.Errorf("Int1024(%#0256x).Int16() = %#032x, want %#032x", tc.a, got, tc.want)
+		}
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply cast_test.go:1540]: fixed in https://github.com/shogo82148/ints/pull/29/commits/d0cff8efba17da8a97f12d83c9e0ce5534073a52

File: cast_test.go [hunk 5]
```diff
@@ -0,0 +1,1975 @@
...
+			},
+			Int256{M - 1, M - 1, M - 1, M - 1},
+		},
+	}
+
+	for _, tc := range testCases {
+		got := tc.a.Int256()
+		if got != tc.want {
+			t.Errorf("Int1024(%#0256x).Int16() = %#064x, want %#064x", tc.a, got, tc.want)
+		}
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply cast_test.go:1579]: fixed in https://github.com/shogo82148/ints/pull/29/commits/d0cff8efba17da8a97f12d83c9e0ce5534073a52

File: cast_test.go [hunk 6]
```diff
@@ -0,0 +1,1975 @@
...
+				M - 1, M - 1, M - 1, M - 1,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		got := tc.a.Int512()
+		if got != tc.want {
+			t.Errorf("Int1024(%#0256x).Int16() = %#0128x, want %#0128x", tc.a, got, tc.want)
+		}
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply cast_test.go:1621]: fixed in https://github.com/shogo82148/ints/pull/29/commits/d0cff8efba17da8a97f12d83c9e0ce5534073a52
